modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/dml/UpdatePlanBuilder.java

```diff
@@ -261,8 +266,16 @@ else if (stmt instanceof GridSqlMerge) {
             else
                 hasValProps = true;
         }
-
-        if (hasKeyProps && !rowKeys.isEmpty()) {
+    
+        rowKeys.removeIf(rowKey -> desc.type().property(rowKey).defaultValue() != null);
+    
+        boolean fillPKsWithDefaultsSysSetting = IgniteSystemProperties.getBoolean(
+                IgniteSystemProperties.IGNITE_SQL_FILL_ABSENT_PK_WITH_DEFAULTS, false);
```

Comment:
Let's cache the default value in class field.
Properties has a HashTable underneath with synchronized methods.
So, using Properties on hot-path is a bad practice.

Reply:
Cached it in another class, thx.